### PR TITLE
[AMORO-3210] Support loading flink mixed catalog without ams url

### LINKED ISSUE #3210: [Subtask]: Support loading mixed catalog without AMS dependency on Flink
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

As titled.

### Parent issue

#3188

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Close #3210.

## Brief change log

- Support loading Flink mixed catalog without ams url
- Support loading Flink unified catalog without ams url
- Remove some duplicated codes in CatalogUtil

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by Aireed — APPROVED** [review]
i left some tiny comment.

**Comment by Aireed** [comment]
Here, the definition of the catalog's property is completely unrestricted. I suggest keeping the original configuration validation. What do others think?  @zhongqishang

File: amoro-format-mixed/amoro-mixed-flink/amoro-mixed-flink-common/src/main/java/org/apache/amoro/flink/table/UnifiedDynamicTableFactory.java [hunk 1]
```diff
@@ -42,12 +42,12 @@
  * AmoroDynamicTableFactory is a factory for creating dynamic table sources and sinks. It implements
```
[the remark this answers is not in the document]
Comment (zhoujinsong) [reply amoro-format-mixed/amoro-mixed-flink/amoro-mixed-flink-common/src/main/java/org/apache/amoro/flink/table/UnifiedDynamicTableFactory.java]: Fixed.

File: amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/BasicMixedCatalog.java [hunk 2]
```diff
@@ -214,7 +215,7 @@ public Map<String, String> properties() {
   protected TableMeta getMixedTableMeta(TableIdentifier identifier) {
     TableMeta tableMeta;
     try {
-      tableMeta = getClient().getTable(MixedFormatCatalogUtil.amsTaleId(identifier));
+      tableMeta = getClient().getTable(CatalogUtil.amsTaleId(identifier));
```
[the remark this answers is not in the document]
Comment (zhoujinsong) [reply amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/BasicMixedCatalog.java]: Fixed.

**Comment by zhoujinsong** [comment]
> Here, the definition of the catalog's property is completely unrestricted. I suggest keeping the original configuration validation. What do others think? @zhongqishang

Thanks for the review, I have made some changes based on the comments.
I removed the catalog properties limit from the catalog factory.
I hope to open up more catalog properties at this stage. I feel that it may not be possible to list all the catalog properties we wish to support for the time being.

And I found the same implementation in Iceberg and Paimon Flink catalog factory:
https://github.com/apache/iceberg/blob/main/flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java#L133
https://github.com/apache/paimon/blob/master/paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalogFactory.java#L47
